### ISSUE TITLE
`Self` is not bound for a classmethod in a `ParamSpec` protocol

### ISSUE THREAD

**Issue by jamtat** [astral-sh/ty#4369]
### Summary

Ty rejects a concrete class that implements a protocol generic over a `ParamSpec` when a classmethod returns `Self`.

```python
from typing import Protocol, Self


class FactoryP[**P](Protocol):
    @classmethod
    def bind(cls, *args: P.args, **kwargs: P.kwargs) -> Self: ...


class Factory:
    @classmethod
    def bind(cls, value: int) -> Self:
        return cls()


factory: type[FactoryP[[int]]] = Factory
```

Expected: no diagnostic. `Self` should bind to `Factory` during structural protocol matching.

Actual with Ty 0.0.74:

```text
error[invalid-assignment]: Object of type `<class 'Factory'>` is not assignable to `type[FactoryP[(int, /)]]`
info: incompatible return types: `Factory` is not assignable to `Self@bind`
```

Using `def bind[T](cls: type[T], ...) -> T` makes Ty pass, but Ruff reports `PYI019` and replaces it with the failing `Self` form with `--fix` enabled. 

### Version

0.0.74
